gsa/src/web/entities/filterprovider.js

```diff
@@ -0,0 +1,130 @@
+/* Copyright (C) 2019 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import React, {useEffect} from 'react';
+
+import {useSelector, useDispatch} from 'react-redux';
+
+import Filter from 'gmp/models/filter';
+
+import {isDefined} from 'gmp/utils/identity';
+
+import Loading from 'web/components/loading/loading';
+
+import getPage from 'web/store/pages/selectors';
+import {loadUserSettingDefault} from 'web/store/usersettings/defaults/actions';
+import {getUserSettingsDefaults} from 'web/store/usersettings/defaults/selectors';
+import {loadUserSettingsDefaultFilter} from 'web/store/usersettings/defaultfilters/actions';
+import {getUserSettingsDefaultFilter} from 'web/store/usersettings/defaultfilters/selectors';
+
+import PropTypes from 'web/utils/proptypes';
+import withGmp from 'web/utils/withGmp';
+
+const ROWS_PER_PAGE_SETTING_ID = '5f5a8712-8017-11e1-8556-406186ea4fc5';
+export const DEFAULT_FALLBACK_FILTER = Filter.fromString('sort=name first=1');
+
+const FilterProvider = ({
+  children,
+  fallbackFilter,
+  gmp,
+  gmpname,
+  locationQuery = {},
+}) => {
+  const dispatch = useDispatch();
+
+  let returnedFilter;
+
+  let defaultSettingFilter = useSelector(state =>
+    getUserSettingsDefaultFilter(state, gmpname).getFilter(),
+  );
+
+  useEffect(() => {
+    if (!isDefined(defaultSettingFilter)) {
+      dispatch(loadUserSettingsDefaultFilter(gmp)(gmpname));
```

Comment:
I would love to replace this with something like `loadUserSettingsDefaultFilterIfNeeded` which checks the state itself if the setting has already been loaded. But this is for another PR.